pkg/acsengine/engine.go

```diff
@@ -283,10 +291,26 @@ func prepareTemplateFiles(properties *api.Properties) ([]string, string, error)
 	return files, baseFile, nil
 }
 
-func getParameters(properties *api.Properties, isClassicMode bool) (map[string]interface{}, error) {
+//GetCloudSpecConfig returns the kubenernetes container images url configurations based on the deploy target environment
+//for example: if the target is the public azure, then the default container image url should be gcr.io/google_container/...
+//if the target is azure china, then the default container image should be mirror.azure.cn:5000/google_container/...
+func GetCloudSpecConfig(location string) AzureEnvironmentSpecConfig {
+	cloudSpecConfig := AzureCloudSpec
+	switch location {
```

Comment:
Is this going to silently ignore someone who tries to use `AzureGermanCloud`?

Seems like they're silently going to get PublicCloud? Do we need a validation check?